core/control-plane/contract-core/src/test/java/org/eclipse/edc/connector/contract/offer/ContractOfferResolverImplIntegrationTest.java

```diff
@@ -107,6 +102,35 @@ void shouldLimitResult_withHeterogenousChunks() {
         verify(policyStore).findById("contract");
     }
 
+    @Test
+    void shouldReturnOfferSubsetAcrossMultipleContractDefinitions() {
+
+        var assets1 = range(1, 11).mapToObj(i -> createAsset("asset-" + i).build()).collect(Collectors.toList());
+        var assets2 = range(11, 21).mapToObj(i -> createAsset("asset-" + i).build()).collect(Collectors.toList());
+
+        store(assets1);
+        store(assets2);
+
+        var contractDefinition1 = getContractDefBuilder("contract-definition-")
+                .selectorExpression(selectorFrom(assets1)).build();
+        var contractDefinition2 = getContractDefBuilder("contract-definition-")
+                .selectorExpression(selectorFrom(assets2)).build();
+
+        when(agentService.createFor(isA(ClaimToken.class))).thenReturn(new ParticipantAgent(emptyMap(), emptyMap()));
+        when(contractDefinitionService.definitionsFor(isA(ParticipantAgent.class))).thenAnswer(i -> Stream.of(contractDefinition1, contractDefinition2));
+        when(policyStore.findById(any())).thenReturn(PolicyDefinition.Builder.newInstance().policy(Policy.Builder.newInstance().build()).build());
+
+        var from = 0;
+        var to = 12;
+        var query = ContractOfferQuery.builder().range(new Range(from, to)).claimToken(ClaimToken.Builder.newInstance().build()).build();
+        assertThat(contractOfferResolver.queryContractOffers(query)).hasSize(12);
+
+        from = 8;
+        to = 14;
+        query = ContractOfferQuery.builder().range(new Range(from, to)).claimToken(ClaimToken.Builder.newInstance().build()).build();
+        assertThat(contractOfferResolver.queryContractOffers(query)).hasSize(6);
```

Comment:
nit: may want to use variables here, i.e. `to - from`

Reply:
ty, fixed.